### DISCUSSION TITLE
`observable.shallow` doesn't react in `reaction`.

### DISCUSSION THREAD

**Discussion by ygnoh** [mobxjs/mobx#2929] [Q&A] [answered]
Hi all, I wonder why this reaction doesn't react:

```js
const a = { name: "hi" };

class B {
  _as = [];

  constructor() {
    makeObservable(this, {
      _as: observable.shallow
    });
  }

  get as() {
    return this._as;
  }

  push(a) {
    this._as.push(a);
  }
}

const b = new B();

reaction(
  () => b.as,
  () => console.log("changed")
);

console.log("start pushing");
b.push(a);
console.log("end pushing");
```

I thought because `_as` is `observable.shallow` and it changed, the `reaction` will react. but it doesn't.

`observable.shallow` doesn't work in `reaction`? and how can I fix it to work as expected?

Thank you for considering my question.

---

You can see this code also in codesandbox: https://codesandbox.io/s/pure-mobx-nfvp1?file=/src/index.js

**Comment by mweststrate** [comment, marked as answer]
`b.as` is a reference that doesn't change, so there is no reason to react here. You are probably looking for `() => b.as.length` (for example). Make sure to grok https://mobx.js.org/understanding-reactivity.html

**Reply by ygnoh** [reply]
Thank you for answering
